web/src/beta/features/PluginPlayground/Plugins/usePlugins.ts

```diff
@@ -260,39 +267,33 @@ export default () => {
   }, [selectedPlugin, setNotification]);
 
   const encodeAndSharePlugin = useCallback(
-    (pluginId: string): string | undefined => {
-      selectPlugin(pluginId);
-      // Need to "find" the selectedPlugin as it does not get updated immediately
-      const sharedPlugin = plugins.find((plugin) => plugin.id === pluginId);
-
-      // Note: We can't use the same id for a shared plugin
-      const selectedPluginCopy = { ...sharedPlugin };
-
-      // First compress the code
+    (pluginId: string) => {
       try {
+        const pluginToShare = plugins.find((plugin) => plugin.id === pluginId);
+        const selectedPluginCopy = { ...pluginToShare, id: SHARED_PLUGIN_ID };
+
         const compressed = LZString.compressToBase64(
           JSON.stringify(selectedPluginCopy)
         )
-          .replace(/\+/g, "-") // Convert + to -
-          .replace(/\//g, "_") // Convert / to _
-          .replace(/=/g, ""); // Remove padding =
+          .replace(/\+/g, "-")
+          .replace(/\//g, "_")
+          .replace(/=/g, "");
 
-        const shareUrl = `${window.location.origin}${window.location.pathname}?plugin=${compressed}`;
+        const shareUrl = `${window.location.origin}${window.location.pathname}?shared-plugin-url=${compressed}`;
```

Comment:
```suggestion
        const shareUrl = `${window.location.origin}${window.location.pathname}?shared=${compressed}`;
```
Could we just use `shared-plugin` or `shared`, the value is base64 string not url.

Reply:
done